gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/ImmunizationMapper.java

```diff
@@ -0,0 +1,252 @@
+            if (ts.getValue() != null) {
+                return ts.getValue();
+            } else if (ts.getNullFlavor().equals(CsNullFlavor.UNK)) {
+                return CsNullFlavor.UNK.value();
+            }
+        }
+
+        return null;
+    }
+
+    private String getIVXBTSStringValue(IVXBTS ivxbts) {
+        if (ivxbts != null) {
+            if (ivxbts.getValue() != null) {
+                return ivxbts.getValue();
+            } else if (ivxbts.getNullFlavor().equals(CsNullFlavor.UNK)) {
+                return CsNullFlavor.UNK.value();
+            }
+        }
+
+        return null;
+    }
+
+    private DateTimeType getObservationDate(RCMRMT030101UK04ObservationStatement observationStatement) {
+        var effectiveTime = observationStatement.getEffectiveTime();
+        if (effectiveTime != null) {
+            var center = getTSStringValue(effectiveTime.getCenter());
+            var low = getIVXBTSStringValue(effectiveTime.getLow());
+            var high = getIVXBTSStringValue(effectiveTime.getHigh());
+            var availabilityTimeValue = observationStatement.getAvailabilityTime();
+
+            if (center != null) {
+                return DateFormatUtil.parseToDateTimeType(center);
+            } else if (low != null && high != null) {
+                return DateFormatUtil.parseToDateTimeType(low);
+            } else if (low != null) {
+                return DateFormatUtil.parseToDateTimeType(low);
+            } else if (high != null && availabilityTimeValue.getValue() != null && availabilityTimeValue.getNullFlavor() == null) {
+                return DateFormatUtil.parseToDateTimeType(availabilityTimeValue.getValue());
+            }
+        }
+
+        return null;
+    }
+
+    private String createDateText(RCMRMT030101UK04ObservationStatement observationStatement) {
+        if (observationStatement.getEffectiveTime() != null) {
+            var low = getIVXBTSStringValue(observationStatement.getEffectiveTime().getLow());
+            var high = getIVXBTSStringValue(observationStatement.getEffectiveTime().getHigh());
+            var availabilityTimeValue = observationStatement.getAvailabilityTime();
+
+            if (low != null && high != null) {
+                return END_DATE_PREFIX + high;
+            } else if (high != null && availabilityTimeValue.getValue() != null) {
+                return END_DATE_PREFIX + high;
+            } else if (high != null && availabilityTimeValue.getNullFlavor() != null) {
+                return END_DATE_PREFIX + high;
+            }
+        }
+
+        return null;
+    }
+
+    private Annotation buildNote(RCMRMT030101UK04ObservationStatement observationStatement) {
+        var pertinentText = getPertinentAnnotation(observationStatement);
+        var dateText = createDateText(observationStatement);
+
+        if (pertinentText != null && dateText != null) {
+            return new Annotation(new StringType(pertinentText + StringUtils.SPACE + dateText));
```

Comment:
According to the spreadsheet you need to be able to handle all instances of pertinentInformation/pertinentAnnotation/text it will look something like this:
```suggestion
            observationStatement
            .getPertinentInformation()
            .stream()
            .map(RCMRMT030101UK04PertinentInformation02::getPertinentAnnotation)
            .map(RCMRMT030101UK04Annotation::getText)
            .map(StringType::new)
            .map(Annotation::new)
            .collect(Collectors.toList());
```